backend/src/main/java/saci/domain/service/LevelService.java

```diff
@@ -40,4 +40,16 @@ public Level createLevel(Level level) {
         }
         return levelRepository.save(level);
     }
+
+    public Optional<Level> findLevelByScore(Long roleId, double score) {
+        return levelRepository
+                .findByRoleIdAndMinCoefficientLessThanEqualAndMaxCoefficientGreaterThanEqual(
```

Comment:
Maybe instead of using this confusing train variable it's better to use @Query annotation also from JPA.